### PR TITLE
feat(workflow): Dynamic counts guide anchor for external release

### PR DESCRIPTION
**Pull request by taylangocmen**
This adds a guide anchor to the first issue row's dynamic event count. Also reverted the chart stats period selection to 24h from the selected period.

Resolves:
[WOR-145](https://getsentry.atlassian.net/browse/WOR-145) 
[WOR-309](https://getsentry.atlassian.net/browse/WOR-309)

### CODE REVIEW

File: src/sentry/assistant/guides.py [hunk 1]
```diff
@@ -110,4 +110,18 @@
...
+        "steps": [
+            {
+                "title": _("These counts have changed"),
+                "message": _(
+                    "These numbers and the bar chart now respect the time selected and any"
+                    "search filters you've applied. You can hover to see the totals."
+                ),
+                "target": "dynamic_counts",
+            }
+        ],
```
Comment (markstory) [inline src/sentry/assistant/guides.py:125]: I don't think you need the steps/title/message in the server definition of guides. It doesn't get used in the API responses. The other guides need to be cleaned-up still. I can do that separately.

File: src/sentry/assistant/guides.py [hunk 2]
```diff
@@ -110,4 +110,18 @@
             }
         ],
     },
+    "dynamic_counts": {
+        "id": 4,
```
Comment (markstory) [inline src/sentry/assistant/guides.py:114]: We most recently used `id:4` for the discover sidebar guide. You might end up with some overlap if you re-use that id.

Comment (taylangocmen) [reply src/sentry/assistant/guides.py:114]: Should I use 5 instead?

File: src/sentry/api/endpoints/organization_group_index.py [hunk 3]
```diff
@@ -115,13 +115,8 @@ def get(self, request, organization):
         if stats_period not in (None, "", "24h", "14d"):
             return Response({"detail": ERR_INVALID_STATS_PERIOD}, status=400)
         elif stats_period is None:
-            if has_dynamic_issue_counts and start and end:
-                # custom date range
-                stats_period = "auto"
```
[the remark this answers is not in the document]
Comment (taylangocmen) [reply src/sentry/api/endpoints/organization_group_index.py:120]: We still use auto as the secondary stats period option for the charts in the issue list, I'll add it to the valid options in the above if.

File: src/sentry/assistant/guides.py (hunk 1 above)
Comment (taylangocmen) [reply src/sentry/assistant/guides.py:125]: Ok I'll remove them from here

File: src/sentry/assistant/guides.py (hunk 2 above)
Comment (markstory) [reply src/sentry/assistant/guides.py]: I took a look in production and 5 has some usage, but 7 is unused.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Review by davidenwang — APPROVED** [review]

**Pull request merged**